app/Http/Controllers/QuizApiController.php

```diff
@@ -50,6 +53,46 @@ public function create(QuizRequest $request): JsonResponse
         ], 500);
     }
 
+    public function results(ResultsRequest $request): JsonResponse
```

Comment:
Since this method is fairly unique, while it is ofcourse related to quizzes, it's also related to everything else! I would suggest making a dedicated ScoreApiController and moving this method into that